File: JDK11/opentraceability/src/main/java/opentraceability/queries/epcisquerystacktraceitem.java
package opentraceability.queries;

import java.util.List;
import java.util.Map;
import java.util.UUID;
import java.net.URI;
import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.HashMap;

import opentraceability.models.identifiers.*;

class EPCISQueryStackTraceItem {
    public String ID = UUID.randomUUID().toString();
    public LocalDateTime Created = LocalDateTime.now();
    //public HttpStatusCode ResponseStatusCode = null;
    public Integer ResponseStatusCode = null;
    public URI RelativeURL = null;
    public List<Map<String, List<String>>> RequestHeaders = null;
    public List<Map<String, List<String>>> ResponseHeaders = null;
    public String RequestBody = null;
    public String ResponseBody = null;
}

File: Kotlin/OpenTraceability/src/main/kotlin/opentraceability/queries/epcisquerystacktraceitem.kt
package opentraceability.queries

import java.util.*
import opentraceability.models.identifiers.*
import java.net.URI
import java.time.LocalDateTime

class EPCISQueryStackTraceItem {

    var ID: String = UUID.randomUUID().toString()
    var Created: LocalDateTime = LocalDateTime.now()
    //var ResponseStatusCode: HttpStatusCode? = null
    var ResponseStatusCode: Int? = null
    var RelativeURL: URI? = null
    var RequestHeaders: MutableList<MutableMap<String, MutableList<String>>>? = null
    var ResponseHeaders: MutableList<MutableMap<String, MutableList<String>>>? = null
    var RequestBody: String? = null
    var ResponseBody: String? = null

}
